components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/MCPUtils.java

```diff
@@ -492,4 +532,50 @@ public static void handleMCPFailure(MessageContext messageContext, McpResponseDt
         Utils.sendFault(messageContext, responseDto.getStatusCode());
     }
 
+    /**
+     * Get the Gateway server URL from the supported vhosts of the API, prioritizing the value passed in
+     * host header of the request
+     *
+     * @param hostHeader  Host header of the request
+     * @param contextPath Context path of the matched API
+     * @return Gateway server URL
+     */
+    public static String getGatewayServerURL(String hostHeader, String contextPath) {
+        String serverURL = null;
+        String tenantDomain = GatewayUtils.getTenantDomain();
+        Map<String, Map<String, API>> tenantAPIMap = DataHolder.getInstance().getTenantAPIMap();
+        Map<String, API> contextApiMap = tenantAPIMap.get(tenantDomain);
+        if (contextApiMap != null) {
+            API api = contextApiMap.get(contextPath);
+            if (api == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("No matching API found for context: " + contextPath + " in tenant: " + tenantDomain);
+                }
+                return null;
+            }
+
+            List<VHost> gwVhosts = api.getVhosts();
+            if (gwVhosts == null || gwVhosts.isEmpty()) {
+                    return null;
+            }
+
+            if (!StringUtils.isEmpty(hostHeader)) {
+                for (VHost vHost: gwVhosts) {
+                    if (vHost.getHost().equals(hostHeader)) {
+                        serverURL = vHost.getHttpsUrl();
+                        break;
+                    }
+                }
+            }
+
+            // If server URL is not resolved using host header, pick the first vhost as the resource endpoint
+            if (StringUtils.isEmpty(serverURL)) {
+                if (gwVhosts != null && !gwVhosts.isEmpty()) {
```

Comment:
We already check this in line 558 right?

Reply:
fixed in https://github.com/wso2/carbon-apimgt/pull/13361/commits/ffa9706e9d639858e2886dad9b3788673018fcd9